mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -119,6 +120,14 @@ final class EMLDAOptimizer extends LDAOptimizer {
       }
     }
 
+    // create term vertices for empty docs
+    val emptyDocTermVertices: RDD[(VertexId, TopicCounts)] =
+      docs.filter { case (_, termCounts: Vector) =>
+        Vectors.norm(termCounts, p = 1.0) == 0.0 // empty docs has all zero termCounts
+      }.mapValues { case(termCounts: Vector) =>
```

Comment:
This can just be `mapValues(termCounts => ...)`. Actually you don't need the type for the `termCounts` above either, or for `emptyDocTermVertices`, I think. That said, maybe don't change this unless you need to do another pass for another reason. And I suppose the code below does write some of this out too. So this seems OK to me as-is as well.